### PR TITLE
Bugfix/firefox

### PR DESCRIPTION
**Pull request by vee-L**
Adds python3 integration and the bugfix for #76

### CODE REVIEW

File: server/main.py [hunk 1]
```diff
@@ -168,4 +168,4 @@ def hello():
 
 if __name__ == "__main__":
     app.debug = True
-    app.run(host=sys.argv[1], port=int(sys.argv[2]))
+    app.run(host='127.0.0.1', port=5000)
```
Comment (ahankinson) [inline server/main.py]: This changes the functionality of the script to not accept the port and ip as command-line arguments. Is this correct?

File: server/static/js/bargraph.js [hunk 2]
```diff
@@ -184,14 +187,15 @@ var BarGraph = function(selector, width, height)
                 sort.direction = SortDirectionEnum.descending;
             }
             chart(chart.data);
-        });
+        } );
```
Comment (ahankinson) [inline server/static/js/bargraph.js]: Why this space?

File: server/static/js/bargraph.js [hunk 3]
```diff
@@ -184,14 +187,15 @@ var BarGraph = function(selector, width, height)
...
 
     }
 
     // GUI components
     attachPrintButton(selector, d3.select(selector).select("svg")[0][0]);
     attachSortChooser(selector);
 
+
```
Comment (ahankinson) [inline server/static/js/bargraph.js]: Too many line-breaks?

**Comment by vee-L** [comment]
Woops, I closed as you were reviewing. I noticed the port issue, I'm going to change that and fix the spacing.

**Comment by ahankinson** [comment]
Why did you close this?

**Comment by ahankinson** [comment]
👍

**Comment by musicus** [comment]
Does it still work with Python 2?

**Comment by vee-L** [comment]
I just ran it from a python 2.7 virtualenv and it seems like it's backwards compatible, no errors are getting spit out while trying out all the functionalities. It's interesting because I don't think python3 is supposed to be backwards compatible, probably the things I changed just happened to be that way.

**Comment by vee-L** [comment]
I'm not sure if there are other things I'm not aware of compatibility wise, the code in this pull request runs fine in a python 2.7 environment, but it's python3 code.

**Comment by musicus** [comment]
Great! Thank you. :smile_cat:

**Comment by ahankinson** [comment]
Can you run it with Travis? I'll give you some links... one sec.

**Comment by musicus** [comment]
Cool: yes, of course we want to run it with Travis! 
:bowtie:

**Comment by ahankinson** [comment]
https://docs.travis-ci.com/user/getting-started/
https://docs.travis-ci.com/user/languages/python

An example of how we're doing both JavaScript and Python testing with Cantus:
https://github.com/DDMAL/cantus/blob/develop/.travis.yml

**Comment by vee-L** [comment]
It says "Python builds are not available on the OS X environment", is that something I should worry about?

**Comment by ahankinson** [comment]
What's your .travis.yml file?

**Comment by ahankinson** [comment]
Or are you still just reading the instructions? It won't be built in OSX, it will be built in a linux VM

**Comment by ahankinson** [comment]
You might also have a look at what Travis looks like for Diva: https://travis-ci.org/DDMAL/diva.js

**Comment by vee-L** [comment]
Oh ok. The travis.yml file needs actual tests written, right? Because there doesn't seem to be any testing implemented for this project that I can find. Should that be its own branch/issue or should I also implement it in this pull request?

**Comment by ahankinson** [comment]
Not quite. The .travis.yml file directs a remote machine on how to build and run the unit tests that you write. So:
1. Write some unit tests
2. Make sure they run locally.
3. Set up .travis.yml to run those tests remotely

I think you should add it to this pull request since a contribution shouldn't be 'complete' without a unit test to verify that it actually works.

**Comment by ahankinson** [comment]
Just to be clear, too: You don't need to design your unit tests to run on both Python 2.7 and 3; you write a single test for the code, and then Travis takes care of running it on 2.7 and 3.

**Comment by vee-L** [comment]
I don't have admin access so I can't turn on travis, does that
 mean I have to fork?

**Comment by vee-L** [comment]
Also browser-specific js testing doesn't seem very doable on travis but I'
m looking into it.

**Comment by ahankinson** [comment]
@vee-L level up! +10 in unit testing. (Admin access granted)

**Comment by ahankinson** [comment]
Don't worry about browser specific testing on Travis. You can get it set up locally, though.

**Comment by vee-L** [comment]
I've merged both bugfixes into this and now travis seems to work well. Javascript testing I think could be an issue just like in elvis-db, because so far it's been harder to implement a working test than to fix the actual bug. Could this be merged like this into dev? (unless you see something you don't like in the commits)

**Pull request merged**